src/utils/issue.ts

```diff
@@ -0,0 +1,171 @@
+import { Context } from "../types/context";
+import { Issue, ISSUE_TYPE } from "../types/payload";
+import { getLinkedPullRequests } from "./get-linked-prs";
+
+export function isParentIssue(body: string) {
+  const parentPattern = /-\s+\[( |x)\]\s+#\d+/;
+  return body.match(parentPattern);
+}
+
+export async function getAssignedIssues(context: Context, username: string): Promise<Issue[]> {
+  const payload = context.payload;
+
+  try {
+    return await context.octokit.paginate(
+      context.octokit.issues.listForRepo,
+      {
+        owner: payload.repository.owner.login,
+        repo: payload.repository.name,
+        state: ISSUE_TYPE.OPEN,
+        per_page: 1000,
+      },
+      ({ data: issues }) => issues.filter((issue) => !issue.pull_request && issue.assignee && issue.assignee.login === username)
+    );
+  } catch (err: unknown) {
+    context.logger.fatal("Fetching assigned issues failed!", err);
+    return [];
+  }
+}
+
+export async function addCommentToIssue(context: Context, message: string | null) {
+  let comment = message as string;
+
+  const { payload } = context;
+
+  const issueNumber = payload.issue.number;
+  try {
+    await context.octokit.issues.createComment({
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      issue_number: issueNumber,
+      body: comment,
+    });
+  } catch (e: unknown) {
+    context.logger.fatal("Adding a comment failed!", e);
+  }
+}
+
+//// Pull Requests \\\\
+
+export async function closePullRequest(context: Context, pullNumber: number) {
+  const { repository } = context.payload;
+  try {
+    await context.octokit.rest.pulls.update({
+      owner: repository.owner.login,
+      repo: repository.name,
+      pull_number: pullNumber,
+      state: "closed",
+    });
+  } catch (err: unknown) {
+    context.logger.fatal("Closing pull requests failed!", err);
```

Comment:
I was thinking of saving fatal for emergencies that can be triaged by our team.

For example, a fatal error in the kernel occurs that can affect every partner and user of our system. So as soon as it occurs, it should ping our Telegram bot, which should relay the message in a triage group on Telegram. 

That way our incident response time can be minimal.

Obviously this should only be saved for actual, fatal emergencies. 

I don't think this qualifies.

Reply:
So this should be a feature of the plugin logger and it should be configurable so as to ping the partner's TG and not Ubiquity TG. This will need to be well documented won't it so devs are aware fatal = tg shitstorm 😂

If a partner is running services through plugins and there is a fatal (not including the kernel because that should be ubiquity only) they should be notified in the same way?

Reply:
Our partners shouldn't be expected to be doing maintenance on our product. Obviously that's ideal but that's unrealistic to expect.

Reply:
I agree ideal but unlikely however I meant in the sense that essentially we'd be a service provider like Cloudflare for example and if your services go down then you are notified as a service user, sometimes far later than you want to be.

How is that being addressed for partners if one of the services we provide hits a fatal for example. Manually case-by-case only if it's not fixed within a set time? Or should they also be included in this TG ping, would they have a dedicated channel we could ping maybe?

Reply:
`fatal` could mean incidents that we should resolve, but I think this would belong to the kernel directly since it is only a concern for us and we don't want to implement it inside every plugin.

Reply:
Do you think there should be two loggers, one that does not include fatal and is meant for plugins while the other can have Supabase and Telegram integrations that's more for the kernel?

Reply:
There are 3 entities here:
1. Kernel
2. Core ubiquity plugins
3. 3rd party plugins

Kernel and core ubiquity plugins should share the same logger instance. So if `logger.fatal()` is called in the kernel or core plugin then it's definitely an issue that we should resolve ASAP.

3rd party plugin developers should be responsible for their own plugins and maintain their own logging and monitoring solutions.

There is no need to add logger to the SDK thus allowing 3rd party plugins to use our own logging solution.

It makes sense to create a separate logger package (which is already [created](https://github.com/ubiquity/ubiquibot-logger) as far as I understand) and use it inside the kernel and core plugins.

Reply:
The logger package was just stripped down to work with 3rd party better than with the kernel and core plugins which is why I thought two loggers made sense

So 3rd party use their own and the template should use the basic `logger.warn() => console.warn()` setup it was originally and the logger refactored for Supabase/Telegram posting